### PR TITLE
feat: navigations crud ui [CU-2f1f58j]

### PR DESCRIPTION
**Pull request by CodeVoyager**
## Ticket

https://app.clickup.com/t/2f1f58j

## Summary

What does this PR do/solve? 

- navigation hidden from content manager
- navigation management from plugins view

## Test Plan

- click on "Manage" button opens main modal
- modal initially display list of all navigations
- admin is able to edit existing entries
- admin is able to create new entry
- admin is able to delete one or multiple navigations at once

### CODE REVIEW

File: admin/src/pages/View/components/NavigationManager/index.tsx [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  const {
+    items = [],
+    handleNavigationsDeletion,
+    handleSubmitNavigation,
+    hardReset,
+  } = useDataManager();
+  const navigations = useMemo(
+    () =>
+      items
+        .map((x: Navigation) => x)
```
Comment (ltsNotMike) [inline admin/src/pages/View/components/NavigationManager/index.tsx:45]: I'm might be missing something but I don't think this is necessary. 🤔

File: admin/src/pages/View/components/NavigationManager/index.tsx [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        .sort((a: Navigation, b: Navigation) => Number(a.id) - Number(b.id)),
```
Comment (ltsNotMike) [inline admin/src/pages/View/components/NavigationManager/index.tsx:46]: I think we could use lodash
```suggestion
    () => _.sortBy(items, 'id')
```

File: admin/src/pages/View/components/NavigationManager/Edit/index.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  const onChange = useCallback(
+    (updated: Navigation) => {
+      setState({
+        view: "EDIT",
+        alreadyUsedNames,
+        current: updated,
+        navigation: initialValue,
+      });
+    },
+    [setState]
```
Comment (ltsNotMike) [inline admin/src/pages/View/components/NavigationManager/Edit/index.tsx:33]: Maybe could be
```suggestion
    [setState, initialValue, alreadyUsedNames]
```

File: admin/src/pages/View/components/NavigationManager/List/index.tsx [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+              <Td>
+                <BaseCheckbox
+                  onValueChange={handleSelect(
+                    navigation,
+                    currentlySelectedSet.has(navigation.id)
+                  )}
+                  value={currentlySelectedSet.has(navigation.id)}
+                />
+              </Td>
+              <Td>{navigation.id}</Td>
```
Comment (ltsNotMike) [inline admin/src/pages/View/components/NavigationManager/List/index.tsx:125]: The font is not aligned with strapi design. Could be like this ... or with "neutral600"
```suggestion
   <Td>
     <Typography textColor="neutral800">
       {navigation.id}
      </Typography>
    </Td>
```

File: admin/src/pages/View/components/NavigationManager/Edit/index.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+}) => {
+  return {
+    startActions: (
+      <Button disabled={state.isLoading} onClick={onReset} variant="tertiary">
+        {getMessage("popup.item.form.button.cancel")}
+      </Button>
+    ),
+    endActions: (
+      <Button disabled={state.isLoading} onClick={onSubmit} variant="secondary">
+        {getMessage("popup.navigation.manage.button.edit")}
```
Comment (ltsNotMike) [inline admin/src/pages/View/components/NavigationManager/Edit/index.tsx:67]: "Edit" is a bit misleading for me. Could be "Save" or maybe "Submit"

File: admin/src/pages/View/components/NavigationManager/index.tsx (hunk 1 above)
Comment (CodeVoyager) [reply admin/src/pages/View/components/NavigationManager/index.tsx:45]: Had runtime errors without it. Read-only array or something like this

File: admin/src/pages/View/components/NavigationManager/Create/index.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  onReset,
+}) => {
+  return {
+    startActions: (
+      <Button disabled={state.isLoading} onClick={onReset} variant="tertiary">
+        {getMessage("popup.item.form.button.cancel")}
+      </Button>
+    ),
+    endActions: (
+      <Button disabled={state.isLoading} onClick={onSubmit} variant="primary">
```
Comment (ltsNotMike) [inline admin/src/pages/View/components/NavigationManager/Create/index.tsx:66]: Should be:
```suggestion
      <Button disabled={state.isLoading} onClick={onSubmit} variant="default">
```

File: admin/src/pages/View/components/NavigationManager/List/index.tsx [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+      <Button
+        disabled={state.isLoading}
+        onClick={() =>
+          setState({
+            view: "CREATE",
+            alreadyUsedNames: navigations.map(({ name }) => name),
+            current: INITIAL_NAVIGATION,
+          })
+        }
+        variant="primary"
```
Comment (ltsNotMike) [inline admin/src/pages/View/components/NavigationManager/List/index.tsx:217]: Should be:
```suggestion
        variant="default"
```

File: admin/src/pages/View/components/NavigationManager/List/index.tsx [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+                })}
+              </Box>
+              <Button onClick={deleteSelected} variant="tertiary">
+                {getMessage("popup.navigation.manage.button.delete")}
+              </Button>
+            </Flex>
+          ) : null}
+        </GridItem>
+      </Grid>
+      <Table rowCount={navigations.concat} colCount={i18nEnabled ? 6 : 5}>
```
Comment (ltsNotMike) [inline admin/src/pages/View/components/NavigationManager/List/index.tsx:97]:
```suggestion
      <Table rowCount={navigations.length} colCount={i18nEnabled ? 6 : 5}>
```

**Review by ltsNotMike — APPROVED** [review]

File: admin/src/pages/View/components/NavigationManager/Create/index.tsx [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        current: updated,
+        alreadyUsedNames,
+      });
+    },
+    [setState]
+  );
+  const validationSchema = useMemo(
+    () => validationSchemaFactory(alreadyUsedNames, formatMessage),
+    [alreadyUsedNames]
+  );
```
Comment (kamilszewczyk0) [inline admin/src/pages/View/components/NavigationManager/Create/index.tsx:42]: Add space between single blocks of code (consts, function) to increase readability

File: admin/src/pages/View/components/NavigationManager/Create/index.tsx [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+        {getMessage("popup.item.form.button.cancel")}
+      </Button>
+    ),
+    endActions: (
+      <Button disabled={state.isLoading} onClick={onSubmit} variant="default">
+        {getMessage("popup.navigation.manage.button.save")}
+      </Button>
+    ),
+  };
+};
```
Comment (kamilszewczyk0) [inline admin/src/pages/View/components/NavigationManager/Create/index.tsx:71]: Move to another file and create generic component

File: admin/src/pages/View/components/NavigationManager/Delete/index.tsx [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+        {getMessage("popup.item.form.button.cancel")}
+      </Button>
+    ),
+    endActions: (
+      <Button disabled={state.isLoading} onClick={onSubmit} variant="danger">
+        {getMessage("popup.navigation.manage.button.delete")}
+      </Button>
+    ),
+  };
+};
```
Comment (kamilszewczyk0) [inline admin/src/pages/View/components/NavigationManager/Delete/index.tsx:54]: Move to another file and create generic component

File: admin/src/pages/View/components/NavigationManager/Edit/index.tsx [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+        {getMessage("popup.item.form.button.cancel")}
+      </Button>
+    ),
+    endActions: (
+      <Button disabled={state.isLoading} onClick={onSubmit} variant="secondary">
+        {getMessage("popup.navigation.manage.button.save")}
+      </Button>
+    ),
+  };
+};
```
Comment (kamilszewczyk0) [inline admin/src/pages/View/components/NavigationManager/Edit/index.tsx:71]: Move to another file and create generic component

File: admin/src/pages/View/components/NavigationManager/Error/index.tsx [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+export const errorFooterActions: FooterActionsFactory = ({ onReset }) => {
+  return {
+    endActions: (
+      <Button onClick={onReset} variant="secondary">
+        {getMessage("popup.navigation.manage.button.goBack")}
+      </Button>
+    ),
+  };
+};
```
Comment (kamilszewczyk0) [inline admin/src/pages/View/components/NavigationManager/Error/index.tsx:43]: Is it possible to use the generic component here as well?

File: admin/src/pages/View/components/NavigationManager/List/index.tsx [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
+            current: INITIAL_NAVIGATION,
+          })
+        }
+        variant="default"
+      >
+        {getMessage("popup.navigation.manage.button.create")}
+      </Button>
+    ),
+  };
+};
```
Comment (kamilszewczyk0) [inline admin/src/pages/View/components/NavigationManager/List/index.tsx:223]: Move to another file and create generic component

File: admin/src/pages/View/components/NavigationManager/List/index.tsx [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+
+const icons = {
+  edit: <Pencil />,
+  delete: <Trash />,
+  visible: <Eye />,
+  notVisible: <EyeStriked />,
+} as const;
```
Comment (kamilszewczyk0) [inline admin/src/pages/View/components/NavigationManager/List/index.tsx:230]: Move to anther file to keep the component clean (there only icons)

File: admin/src/pages/View/components/NavigationManager/Create/index.tsx (hunk 10 above)
Comment (kamilszewczyk0) [inline admin/src/pages/View/components/NavigationManager/Create/index.tsx:71]: Naming of the CRUD components as nouns, not verbs (according to rest of the structure's file naming)

File: admin/src/pages/View/components/NavigationManager/Delete/index.tsx [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+
+const renderItems = (navigations: Array<Navigation>) =>
+  navigations.map(({ name }) => `"${name}"`).join(", ");
```
Comment (kamilszewczyk0) [inline admin/src/pages/View/components/NavigationManager/Delete/index.tsx:57]: Clean up "name" returning from this monster ;)

File: admin/src/pages/View/components/NavigationManager/List/index.tsx [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+      view: "LIST",
+    });
+  };
+  const edit = (navigation: Navigation) => () => {
+    setState({
+      view: "EDIT",
+      navigation,
+      alreadyUsedNames: navigations
+        .map(({ name }) => name)
+        .filter((name) => name !== navigation.name),
```
Comment (kamilszewczyk0) [inline admin/src/pages/View/components/NavigationManager/List/index.tsx:65]: Map and filter changed to reduce (less iterations)

**Review by kamilszewczyk0 — COMMENTED** [review]
Comments are added in places that have been discussed during the meeting

**Comment by cyp3rius** [comment]
@CodeVoyager please do the reworks as separated PR on `master`. As for now merging this because of critical dependency for `2.1.0` release.

**Pull request merged**